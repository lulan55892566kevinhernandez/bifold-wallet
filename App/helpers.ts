export function parseSchema(schemaId?: string): string {
  if (schemaId) {
    const schemaIdRegex = /(.*?):([0-9]):([a-zA-Z .\-_0-9]+):([a-z0-9._\-]+)$/
    const schemaIdParts = schemaId.match(schemaIdRegex)

    if (schemaIdParts!.length === 5) {
      const prettyName = `${schemaIdParts![3].replace(/_/g, ' ')} V${schemaIdParts![4]}`
      return prettyName
    } else {
      return 'Credential'
    }
  } else {
    return 'Credential'
  }
}
