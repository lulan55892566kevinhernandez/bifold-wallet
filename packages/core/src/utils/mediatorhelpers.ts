import { Agent, MediationRecord } from '@credo-ts/core'

export const isMediatorInvitation = async (agent: Agent, url: string): Promise<boolean> => {
  const invitation = await agent.oob.parseInvitation(url)
  if (!invitation) {
    return false
  }
  if (invitation.goalCode === 'aries.vc.mediate') {
    agent.config.logger.info(`Invitation is a mediator invitation with goal code: ${invitation.goalCode}`)
    return true
  }
  agent.config.logger.info(`Invitation is not a mediator invitation, goal code: ${invitation.goalCode}`)
  return false
}

const provisionMediationRecordFromMediatorUrl = async (
  agent: Agent,
  url: string
): Promise<MediationRecord | undefined> => {
  try {
    const invitation = await agent.oob.parseInvitation(url)
    if (!invitation) {
      agent.config.logger.warn(`No invitation found in URL: ${url}`)
      return undefined
    }
    const outOfBandRecord = await agent.oob.findByReceivedInvitationId(invitation.id)
    let [connection] = outOfBandRecord ? await agent.connections.findAllByOutOfBandId(outOfBandRecord.id) : []

    if (!connection) {
      agent.config.logger.warn(`No connection found for out-of-band record: ${outOfBandRecord?.id}`)
      const invite = await agent.oob.parseInvitation(url)
      const { connectionRecord: newConnection } = await agent.oob.receiveInvitation(invite)
      if (!newConnection) {
        agent.config.logger.warn(`Failed to create connection from invitation: ${JSON.stringify(invite, null, 2)}`)
        return
      }
      connection = newConnection
    }
    const result = connection.isReady ? connection : await agent.connections.returnWhenIsConnected(connection.id)
    return agent.mediationRecipient.provision(result)
  } catch (error) {
    agent.config.logger.warn(`Failed to get connection ID from mediator URL: ${error}`)
    return
  }
}

export const setMediationToDefault = async (agent: Agent, mediatorUrl: string) => {
  const mediationRecord = await provisionMediationRecordFromMediatorUrl(agent, mediatorUrl)
  if (!mediationRecord) {
    agent.config.logger.warn(`No connection record found for mediator URL: ${mediatorUrl}`)
    return
  }
  const currentDefault = await agent.mediationRecipient.findDefaultMediator()
  if (currentDefault?.connectionId === mediationRecord.id) {
    agent.config.logger.info(`Default mediator already set for connection ID: ${mediationRecord.id}`)
    return
  }
  await agent.mediationRecipient.setDefaultMediator(mediationRecord)
  agent.config.logger.info(`setting default mediator with record: ${JSON.stringify(mediationRecord)}`)
}
