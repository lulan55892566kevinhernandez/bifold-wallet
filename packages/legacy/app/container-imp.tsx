import { BaseLogger } from '@credo-ts/core'
import { Container, TokenMapping } from '@hyperledger/aries-bifold-core'
import { DependencyContainer } from 'tsyringe'
export class AppContainer implements Container {
  private _container: DependencyContainer
  private log?: BaseLogger

  public constructor(bifoldContainer: Container, log?: BaseLogger) {
    this._container = bifoldContainer.container.createChildContainer()
    this.log = log
  }

  public get container(): DependencyContainer {
    return this._container
  }

  public init(): Container {
    // eslint-disable-next-line no-console
    this.log?.info(`Initializing App container`)
    // Here you can register any component to override components in core package
    // Example: Replacing button in core with custom button
    // this.container.registerInstance(TOKENS.COMP_BUTTON, Button)
    return this
  }

  public resolve<K extends keyof TokenMapping>(token: K): TokenMapping[K] {
    return this._container.resolve(token) as TokenMapping[K]
  }
}
