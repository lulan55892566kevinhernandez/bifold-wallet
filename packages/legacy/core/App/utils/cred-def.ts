import { CredentialMetadataKeys, CredentialExchangeRecord as CredentialRecord } from '@aries-framework/core'

import { credentialSchema, parseSchemaFromId } from './schema'

export function parseCredDefFromId(credDefId?: string, schemaId?: string): string {
  let name = 'Credential'
  if (credDefId) {
    const credDefRegex = /[^:]+/g
    const credDefIdParts = credDefId.match(credDefRegex)
    if (credDefIdParts?.length === 5) {
      name = `${credDefIdParts?.[4].replace(/_|-/g, ' ')}`
        .split(' ')
        .map((credIdPart) => credIdPart.charAt(0).toUpperCase() + credIdPart.substring(1))
        .join(' ')
    }
  }
  if (name.toLocaleLowerCase() === 'default' || name.toLowerCase() === 'credential') {
    name = parseSchemaFromId(schemaId).name
  }
  return name
}

function credentialDefinition(credential: CredentialRecord): string | undefined {
  return credential.metadata.get(CredentialMetadataKeys.IndyCredential)?.credentialDefinitionId
}

export function parsedCredDefName(credential: CredentialRecord): string {
  return parseCredDefFromId(credentialDefinition(credential), credentialSchema(credential))
}
