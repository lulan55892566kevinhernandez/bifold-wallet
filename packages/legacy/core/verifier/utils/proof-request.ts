import {
  Agent,
  AgentMessage,
  AutoAcceptProof,
  ProofExchangeRecord,
  ProofRequest,
  V1RequestPresentationMessage,
} from '@aries-framework/core'

import { defaultProofRequestTemplates } from '../constants'
import {
  IndyRequestedAttribute,
  IndyRequestedPredicate,
  ProofRequestTemplate,
  ProofRequestType,
} from '../types/proof-reqeust-template'

const protocolVersion = 'v1'
const domain = 'http://aries-mobile-agent.com'

/*
 * Find Proof Request message in the storage by the given id
 * */
export const findProofRequestMessage = async (agent: Agent, id: string): Promise<ProofRequest | undefined> => {
  const message = await agent.proofs.findRequestMessage(id)
  if (message && message instanceof V1RequestPresentationMessage && message.indyProofRequest) {
    return message.indyProofRequest
  } else {
    return undefined
  }
}

/*
 * Find Proof Request template for provided id
 * */
export const getProofRequestTemplate = (id: string): ProofRequestTemplate | undefined => {
  return defaultProofRequestTemplates.find((template) => template.id === id)
}

/*
 * Build Proof Request data from for provided template
 * */
export const buildProofRequestDataForTemplate = (
  template: ProofRequestTemplate,
  customValues?: Record<string, Record<string, number>>
) => {
  if (template.payload.type === ProofRequestType.Indy) {
    const requestedAttributes: Map<string, IndyRequestedAttribute> = new Map()
    const requestedPredicates: Map<string, IndyRequestedPredicate> = new Map()
    let index = 0

    template.payload.data.forEach((data) => {
      if (data.requestedAttributes?.length) {
        data.requestedAttributes.forEach((requestedAttribute) => {
          requestedAttributes.set(`referent_${index}`, requestedAttribute)
          index++
        })
      }
      if (data.requestedPredicates?.length) {
        data.requestedPredicates.forEach((requestedPredicate) => {
          const customValue =
            customValues && customValues[data.schema] ? customValues[data.schema][requestedPredicate.name] : undefined
          if (requestedPredicate.parameterizable && customValue) {
            requestedPredicates.set(`referent_${index}`, { ...requestedPredicate, predicateValue: customValue })
          } else {
            requestedPredicates.set(`referent_${index}`, requestedPredicate)
          }
          index++
        })
      }
    })
    return {
      indy: {
        name: template.name,
        version: template.version,
        nonce: Date.now().toString(),
        requestedAttributes,
        requestedPredicates,
      },
    }
  }
  if (template.payload.type === ProofRequestType.DIF) {
    return {}
  }
}

export const buildProofRequestDataForTemplateId = (
  templateId: string,
  customPredicateValues?: Record<string, Record<string, number>>
) => {
  const template = getProofRequestTemplate(templateId)
  if (!template) {
    return undefined
  }
  return buildProofRequestDataForTemplate(template, customPredicateValues)
}

export interface CreateProofRequestInvitationResult {
  request: AgentMessage
  proofRecord: ProofExchangeRecord
  invitation: AgentMessage
  invitationUrl: string
}

/*
 * Create connectionless proof request invitation for provided template
 * */
export const createConnectionlessProofRequestInvitation = async (
  agent: Agent,
  templateId: string,
  customPredicateValues?: Record<string, Record<string, number>>
): Promise<CreateProofRequestInvitationResult | undefined> => {
  const proofFormats = buildProofRequestDataForTemplateId(templateId, customPredicateValues)
  if (!proofFormats) {
    return undefined
  }
  const { message: request, proofRecord } = await agent.proofs.createRequest({
    protocolVersion,
    autoAcceptProof: AutoAcceptProof.Always,
    proofFormats,
  })
  const { message: invitation, invitationUrl } = await agent.oob.createLegacyConnectionlessInvitation({
    recordId: proofRecord.id,
    message: request,
    domain,
  })
  return {
    request,
    proofRecord,
    invitation,
    invitationUrl,
  }
}

export interface SendProofRequestResult {
  proofRecord: ProofExchangeRecord
}

/*
 * Build Proof Request for provided template and send it to provided connection
 * */
export const sendProofRequest = async (
  agent: Agent,
  templateId: string,
  connectionId: string,
  customPredicateValues?: Record<string, Record<string, number>>
): Promise<SendProofRequestResult | undefined> => {
  const proofFormats = buildProofRequestDataForTemplateId(templateId, customPredicateValues)
  if (!proofFormats) {
    return undefined
  }
  const proofRecord = await agent.proofs.requestProof({
    protocolVersion,
    connectionId,
    proofFormats,
  })
  return {
    proofRecord,
  }
}

/*
 * Check if the Proof Request template contains at least one predicate
 * */
export const hasPredicates = (record: ProofRequestTemplate): boolean => {
  if (record.payload.type === ProofRequestType.Indy) {
    return record.payload.data.some((d) => d.requestedPredicates && d.requestedPredicates?.length > 0)
  }
  if (record.payload.type === ProofRequestType.DIF) {
    return false
  }
  return false
}

/*
 * Check if the Proof Request template contains parameterizable predicates
 * */
export const isParameterizable = (record: ProofRequestTemplate): boolean => {
  if (record.payload.type === ProofRequestType.Indy) {
    return record.payload.data.some((d) => d.requestedPredicates?.some((predicate) => predicate.parameterizable))
  }
  if (record.payload.type === ProofRequestType.DIF) {
    return false
  }
  return false
}
