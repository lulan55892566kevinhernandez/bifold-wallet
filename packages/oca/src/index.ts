export * as interfaces from './interfaces'
export * as types from './types'
export * as utils from './utils'
export * as formatters from './formatters'
