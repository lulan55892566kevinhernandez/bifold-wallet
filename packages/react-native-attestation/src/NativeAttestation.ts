import type { TurboModule } from 'react-native';
import { TurboModuleRegistry } from 'react-native';

export interface Spec extends TurboModule {
  generateKey(): Promise<string>;
  sha256(stringToHash: string): Promise<Buffer>;
  appleAttestation(keyId: string, challenge: string): Promise<Buffer>;
}

export default TurboModuleRegistry.getEnforcing<Spec>('Attestation');
