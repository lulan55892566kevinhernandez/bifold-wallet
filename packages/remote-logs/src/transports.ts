import axios, { AxiosRequestConfig } from 'axios'
import { transportFunctionType } from 'react-native-logs'

export interface RemoteLoggerOptions {
  lokiUrl?: string
  lokiLabels?: Record<string, string>
  autoDisableRemoteLoggingIntervalInMinutes?: number
}

export type LokiTransportProps = {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  msg: any
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  rawMsg: any
  level: {
    severity: number
    text: string
  }
  options?: RemoteLoggerOptions
}

export const lokiTransport: transportFunctionType = (props: LokiTransportProps) => {
  // Loki requires a timestamp with nanosecond precision
  // however Date.now() only returns milliseconds precision.
  const timestampEndPadding = '000000'

  if (!props.options) {
    throw Error('props.options is required')
  }

  if (!props.options.lokiUrl) {
    throw Error('props.options.lokiUrl is required')
  }

  if (!props.options.lokiLabels) {
    throw Error('props.options.labels is required')
  }

  if (!props.options.lokiUrl) {
    throw new Error('Loki URL is missing')
  }

  const { lokiUrl, lokiLabels } = props.options
  const { message, data } = props.rawMsg.pop()
  const payload = {
    streams: [
      {
        stream: {
          job: 'react-native-logs',
          level: props.level.text,
          ...lokiLabels,
        },
        values: [[`${Date.now()}${timestampEndPadding}`, JSON.stringify({ message, data })]],
      },
    ],
  }

  const [credentials, href] = lokiUrl.split('@')
  const [username, password] = credentials.split('//')[1].split(':')
  const protocol = credentials.split('//')[0]
  const axiosConfig: AxiosRequestConfig = {
    method: 'post',
    url: lokiUrl,
    data: payload,
  }

  // If username and password are present
  // in the URL, use them for auth
  if (username && password) {
    axiosConfig.auth = {
      username,
      password,
    }
    axiosConfig.url = `${protocol}//${href}`
  }

  axios(axiosConfig)
    .then((res) => {
      if (res.status !== 204) {
        throw new Error(`Expected Loki to return 204, received ${res.status}`)
      }
    })
    .catch((error) => {
      throw new Error(`Error while sending to Loki, ${error}`)
    })
}
