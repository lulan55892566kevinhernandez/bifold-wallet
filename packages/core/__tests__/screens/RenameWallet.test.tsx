import { render } from '@testing-library/react-native'
import React from 'react'

import { StoreProvider } from '../../src'
import { testIdWithKey } from '../../src/utils/testable'
import { testDefaultState } from '../contexts/store'
import { BasicAppContext } from '../helpers/app'
import RenameWallet from '../../src/screens/RenameWallet'

jest.mock('react-native-permissions', () => require('react-native-permissions/mock'))
jest.mock('react-native-vision-camera', () => {
  return require('../../__mocks__/custom/react-native-camera')
})

jest.mock('react-native-device-info', () => {
  return {
    getVersion: () => 1,
    getBuildNumber: () => 1,
  }
})

describe('RenameWallet Screen', () => {
  test('Check non-onboarding rendering', async () => {
    const tree = render(
      <StoreProvider initialState={testDefaultState}>
        <BasicAppContext>
          <RenameWallet/>
        </BasicAppContext>
      </StoreProvider>
    )

    const SaveButton = await tree.getByTestId(testIdWithKey('Save'))
    const CancelButton = await tree.getByTestId(testIdWithKey('Cancel'))

    expect(SaveButton).not.toBeNull()
    expect(CancelButton).not.toBeNull()
    expect(tree).toMatchSnapshot()
  })
})
