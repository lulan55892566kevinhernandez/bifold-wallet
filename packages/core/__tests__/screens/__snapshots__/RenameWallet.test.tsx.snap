// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RenameWallet Screen Check non-onboarding rendering 1`] = `
<RNCSafeAreaView
  edges={
    Object {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "off",
    }
  }
  style={
    Object {
      "backgroundColor": "#000000",
      "flex": 1,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        Object {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <RCTScrollView
      contentContainerStyle={
        Object {
          "flexGrow": 1,
        }
      }
      keyboardShouldPersistTaps="handled"
    >
      <View>
        <View
          style={
            Object {
              "backgroundColor": "#000000",
              "flex": 1,
              "justifyContent": "space-between",
              "padding": 16,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "width": "100%",
              }
            }
          >
            <
              height={100}
              style={
                Object {
                  "marginVertical": 16,
                }
              }
            />
            <Text
              maxFontSizeMultiplier={2}
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": 18,
                    "fontWeight": "normal",
                  },
                  Object {
                    "marginBottom": 16,
                    "width": "100%",
                  },
                ]
              }
            >
              NameWallet.ThisIsTheName
            </Text>
            <View
              style={
                Object {
                  "width": "100%",
                }
              }
            >
              <View
                style={
                  Object {
                    "marginVertical": 10,
                    "width": "100%",
                  }
                }
              >
                <Text
                  maxFontSizeMultiplier={2}
                  style={
                    Array [
                      Object {
                        "color": "#FFFFFF",
                        "fontSize": 18,
                        "fontWeight": "normal",
                      },
                      Object {
                        "marginBottom": 5,
                      },
                    ]
                  }
                >
                  NameWallet.NameYourWallet
                </Text>
                <TextInput
                  accessibilityLabel="NameWallet.NameYourWallet"
                  defaultValue="test-wallet"
                  maxFontSizeMultiplier={2}
                  onBlur={[Function]}
                  onChangeText={[Function]}
                  onFocus={[Function]}
                  selectionColor="#42803E"
                  style={
                    Array [
                      Object {
                        "backgroundColor": "#000000",
                        "borderColor": "#FFFFFFFF",
                        "borderRadius": 4,
                        "borderWidth": 2,
                        "color": "#FFFFFF",
                        "fontSize": 16,
                        "padding": 10,
                      },
                      false,
                    ]
                  }
                  testID="com.ariesbifold:id/NameInput"
                />
                <Text
                  maxFontSizeMultiplier={2}
                  style={
                    Array [
                      Object {
                        "color": "#FFFFFF",
                        "fontSize": 18,
                        "fontWeight": "normal",
                      },
                      Object {
                        "alignSelf": "flex-end",
                        "color": "#FFFFFF",
                      },
                    ]
                  }
                >
                  11
                  /
                  50
                </Text>
              </View>
            </View>
          </View>
          <View
            style={Object {}}
          >
            <View
              style={
                Object {
                  "width": "100%",
                }
              }
            >
              <View
                accessibilityLabel="Global.Save"
                accessibilityRole="button"
                accessibilityState={
                  Object {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": false,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  Object {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "backgroundColor": "#42803E",
                    "borderRadius": 4,
                    "opacity": 1,
                    "padding": 16,
                  }
                }
                testID="com.ariesbifold:id/Save"
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "center",
                    }
                  }
                >
                  <Text
                    maxFontSizeMultiplier={2}
                    style={
                      Array [
                        Object {
                          "color": "#FFFFFF",
                          "fontSize": 18,
                          "fontWeight": "normal",
                        },
                        Array [
                          Object {
                            "color": "#FFFFFF",
                            "fontSize": 18,
                            "fontWeight": "bold",
                            "textAlign": "center",
                          },
                          false,
                          false,
                          false,
                        ],
                      ]
                    }
                  >
                    Global.Save
                  </Text>
                </View>
              </View>
              <View
                style={
                  Object {
                    "marginTop": 8,
                  }
                }
              >
                <View
                  accessibilityLabel="Global.Cancel"
                  accessibilityRole="button"
                  accessibilityState={
                    Object {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": false,
                      "expanded": undefined,
                      "selected": undefined,
                    }
                  }
                  accessibilityValue={
                    Object {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "borderColor": "#42803E",
                      "borderRadius": 4,
                      "borderWidth": 2,
                      "opacity": 1,
                      "padding": 16,
                    }
                  }
                  testID="com.ariesbifold:id/Cancel"
                >
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "justifyContent": "center",
                      }
                    }
                  >
                    <Text
                      maxFontSizeMultiplier={2}
                      style={
                        Array [
                          Object {
                            "color": "#FFFFFF",
                            "fontSize": 18,
                            "fontWeight": "normal",
                          },
                          Array [
                            Object {
                              "color": "#42803E",
                              "fontSize": 18,
                              "fontWeight": "bold",
                              "textAlign": "center",
                            },
                            false,
                            false,
                            false,
                          ],
                        ]
                      }
                    >
                      Global.Cancel
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</RNCSafeAreaView>
`;
